dae/dae/genomic_resources/fsspec_protocol.py

```diff
@@ -219,6 +221,42 @@ def __init__(
 
         self.filesystem.makedirs(self.url, exist_ok=True)
 
+    def _get_resource_file_lockfile_path(
+        self, resource: GenomicResource, filename: str
+    ) -> str:
+        """Return path of the resource file's lockfile."""
+        if self.scheme != "file":
+            raise NotImplementedError()
+        resource_url = self.get_resource_url(resource)
+        path = os.path.join(resource_url, ".grr", f"{filename}.lockfile")
+        return path.removeprefix(f"{self.scheme}://")
+
+    def obtain_resource_file_lock(
+        self, resource: GenomicResource, filename: str
+    ):
+        """Lock a resource's file."""
+        class Lock:
+            def __enter__(self):
+                pass
+
+            def __exit__(self, *args):
+                pass
+
+        lock = Lock()
+
+        if self.scheme == "file":
+            path = self._get_resource_file_lockfile_path(resource, filename)
+            if not self.filesystem.exists(os.path.dirname(path)):
+                self.filesystem.makedirs(
+                    os.path.dirname(path), exist_ok=True)
+            lockfile = open(path, "wt", encoding="utf8")
+            lockfile.write(str(datetime.datetime.now()) + "\n")
+            fcntl.flock(lockfile, fcntl.LOCK_EX)
+            lock.__enter__ = lockfile.__enter__
+            lock.__exit__ = lockfile.__exit__
+
+        return lock
+
```

Comment:
I am not sure, but it seems to me that it would be better to raise `NotImplementedError` if we do not support locking. Probably it needs discussion.